sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringExpressions.scala

```diff
@@ -461,68 +462,249 @@ case class FindInSet(left: Expression, right: Expression) extends BinaryExpressi
 }
 
 /**
- * A function that trim the spaces from both ends for the specified string.
+ * A function that trim the spaces or a trim string from both ends for the specified string.
  */
 @ExpressionDescription(
-  usage = "_FUNC_(str) - Removes the leading and trailing space characters from `str`.",
+  usage = """
+    _FUNC_(str) - Removes the leading and trailing space characters from `str`.
+    _FUNC_(BOTH trimString FROM str) - Remove the leading and trailing trimString from `str`
+    _FUNC_(LEADING trimChar FROM str) - Remove the leading trimString from `str`
+    _FUNC_(TRAILING trimChar FROM str) - Remove the trailing trimString from `str`
+  """,
   extended = """
+    Arguments:
+      str - a string expression
+      trimString - the trim string
+      BOTH, FROM - these are keyword to specify for trim string from both side of the string
+      LEADING, FROM - these are keyword to specify for trim string from left side of the string
+      TRAILING, FROM - these are keyword to specify for trim string from right side of the string
     Examples:
       > SELECT _FUNC_('    SparkSQL   ');
        SparkSQL
+      > SELECT _FUNC_(BOTH 'SL' FROM 'SSparkSQLS');
```

Comment:
According to the docs in pr description, Oracle and DB2 both require that trimCharacter can only be 1 character, while MySQL allows multiple characters. Is there any reason we follow MySQL instead of Oracle and DB2?